ioc_copier/ioc_copier.py

```diff
@@ -0,0 +1,104 @@
+        for folder in subFolder:
+            if f"IOC_{start_copy}" in folder:
+                os.rename(os.path.join(root, folder), os.path.join(root, folder.replace(f"IOC_{start_copy}",
+                                                                                        f"IOC_{current_copy}")))
+            if f"IOC-{start_copy}" in folder:
+                os.rename(os.path.join(root, folder), os.path.join(root, folder.replace(f"IOC-{start_copy}",
+                                                                                        f"IOC-{current_copy}")))
+
+os.chdir(os.path.join(os.getcwd(), "iocBoot"))
+for current_copy in range(initial_copy, max_copy+1):
+    copytree(os.path.join(os.getcwd(), f"ioc{ioc_name}-{start_copy}"),
+             os.path.join(os.getcwd(), f"ioc{ioc_name}-{current_copy}"))
+    path = os.path.join(os.getcwd(), f"ioc{ioc_name}-{current_copy}")
+    for root, subFolder, files in os.walk(path):
+        for file in files:
+            if "exe" not in file and "obj" not in file and "pdb" not in file:
+                with open(os.path.join(root, file), "r") as fp:
+                    print(os.path.join(root, file))
+                    text = fp.readlines()
+
+                for i in range(len(text)):
+                    temp = re.sub(f"IOC_{start_copy}", f"IOC_{current_copy}",text[i])
+                    text[i] = temp
+                    temp = re.sub(f"IOC-{start_copy}", f"IOC-{current_copy}",text[i])
+                    text[i] = temp
+                    temp = re.sub(f"{ioc}_{start_copy}", f"{ioc}_{current_copy}", text[i])
+                    text[i] = temp
+                    temp = re.sub(f"RAMPFILELIST{start_copy}", f"RAMPFILELIST{current_copy}", text[i])
+                    text[i] = temp
+                with open(os.path.join(root, file), "w") as fp:
+                    fp.seek(0)
+                    fp.writelines(text)
+                    fp.truncate()
+            if f"IOC_{start_copy}" in file:
+                os.rename(os.path.join(root, file), os.path.join(root, file.replace(f"IOC_{start_copy}",
+                                                                                    f"IOC_{current_copy}")))
+            if f"IOC-{start_copy}" in file:
+                os.rename(os.path.join(root, file), os.path.join(root, file.replace(f"IOC-{start_copy}",
+                                                                                    f"IOC-{current_copy}")))
+        for folder in subFolder:
+            if f"IOC_{start_copy}" in folder:
+                os.rename(os.path.join(root, folder), os.path.join(root, folder.replace(f"IOC_{start_copy}",
+                                                                                        f"IOC_{current_copy}")))
+            if f"IOC-{start_copy}" in folder:
+                os.rename(os.path.join(root, folder), os.path.join(root, folder.replace(f"IOC-{start_copy}",
```

Comment:
Similarly to lines 30-61, could this be broken up into functions to compartmentalise functionality a bit more whilst also increasing readability with descriptive docstrings.